www/src/pages/en/usage/trpc.md

```diff
@@ -143,6 +144,91 @@ Compare this to the tRPC example above and you can see some of the advantages of
 - You don’t need to validate that the request query or body contains the correct data in the procedure, because Zod takes care of this.
 - Instead of creating a response, you can throw errors and return a value or object as you would in any other TypeScript function.
 
+## Useful snippets
+
+Here are some useful snippets that you might find useful for certain use cases.
+
+### Enabling CORS
+
+If you need to consume your API from a different domain, say you create a monorepo and add a Vite app, you might need to enable CORS:
+
+```ts
+// pages/api/trpc/[trpc].ts
+import type { NextApiRequest, NextApiResponse } from "next";
+import { createNextApiHandler } from "@trpc/server/adapters/next";
+import { appRouter, createContext } from "";
+import cors from "nextjs-cors";
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  // Enable cors
+  await cors(req, res);
+
+  // Create and call the tRPC handler
+  return createNextApiHandler({
+    router: appRouter,
+    createContext: createContext,
+  })(req, res);
+};
+
+export default handler;
+```
+
+### Optimistic updates
+
+Optimistic updates are when we update the UI before the API call has finished. This gives the user a better experience, because they don't have to wait for the API call to finish before they can see the result of their action. You can read more on the [React Query docs](https://tanstack.com/query/v4/docs/guides/optimistic-updates).
```

Comment:
Maybe add a sentence about the downside - your app is temporarily showing the user some data that might not be true.